### PR TITLE
Added featurePolicies e2e tests

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

Added featurePolicies e2e tests

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: e2e/cloud_manager/feature_policies_test.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+		externalManagSystemName = policy.ExternalManagementSystem.Name
+	})
+
+	t.Run("Update", func(t *testing.T) {
+		cmd := exec.Command(cliPath,
+			entity,
+			featurePolicies,
+			"update",
+			"--name",
+			externalManagSystemName,
```
Comment (gssbzn) [inline e2e/cloud_manager/feature_policies_test.go:72]: I'm I bit worry about running this in parallel (multiple tests) as if we may not be getting as response from the current test but from a parallel one
I think the `externalManagSystemName` doesn't matter so we could do these as random values
@colm-quinn what do you think?

**Review by gssbzn — DISMISSED** [review]
LGTM but let's hear from colm in case we can improve this a littel

File: e2e/cloud_manager/feature_policies_test.go (hunk 1 above)
Comment (colm-quinn) [reply e2e/cloud_manager/feature_policies_test.go:72]: > I'm I bit worry about running this in parallel (multiple tests) as if we may not be getting as response from the current test but from a parallel one

Good spot. Last write wins with the endpoint, so if you're running in parallel I'd suggest creating a randomised group, so you can ensure the update is only from your test call.

> I think the externalManagSystemName doesn't matter so we could do these as random values
@colm-quinn what do you think?

Yes to randomised names so we can validate, however, in reference to my point above, that will just ensure the correct update, but still be potentially flaky.

File: e2e/cloud_manager/feature_policies_test.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+		var policy *opsmngr.FeaturePolicy
+		if err := json.Unmarshal(resp, &policy); err != nil {
+			t.Fatalf("unexpected error: %v", err)
+		}
+
+		if len(policy.Policies) == 0 {
+			t.Error("No policies found")
+		}
+
```
Comment (colm-quinn) [inline e2e/cloud_manager/feature_policies_test.go:95]: [q] What about ensuring only the requested policies were written.

```suggestion
if len(policy.Policies) != 2 {
			t.Error("Policy count mismatch")
		}
```

File: e2e/cloud_manager/feature_policies_test.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+	t.Run("Update", func(t *testing.T) {
+		cmd := exec.Command(cliPath,
+			entity,
+			featurePolicies,
+			"update",
+			"--name",
+			externalManagSystemName,
+			"--policy",
+			policyExternallyManagedLock,
```
Comment (colm-quinn) [inline e2e/cloud_manager/feature_policies_test.go:74]: [q] How do you specify the `disabledParams` for a policy with this command line? Those are needed for the `DISABLE_SET_MONGOD_CONFIG` policy in particular and should be settable.

**Review by colm-quinn — CHANGES_REQUESTED** [review]
I agree we should make sure parallel tests don't impact on each other.

Also I'd like to see disabledParams tested for the DISABLE_SET_MONGOD_CONFIG, as that is a use case for the operator and it would be useful if the cli can update the params.
 
If that feature doesn't work, then we should open a ticket.

File: e2e/cloud_manager/feature_policies_test.go (hunk 3 above)
Comment (andreaangiolillo) [reply e2e/cloud_manager/feature_policies_test.go:74]: we actually don't allow the user to set `disabledParams` from the command line. I am not sure if we want to add this flag as part of this PR though. @gssbzn what do you think?

Comment (gssbzn) [reply e2e/cloud_manager/feature_policies_test.go:74]: Let's open a ticket and bring it for planning, let's focus on test for this PR

Comment (colm-quinn) [reply e2e/cloud_manager/feature_policies_test.go:63]: Yup, def a separate ticket!

**Review by gssbzn — APPROVED** [review]
LGTM thanks for those fixes

File: e2e/cloud_manager/feature_policies_test.go (hunk 3 above)
Comment (andreaangiolillo) [reply e2e/cloud_manager/feature_policies_test.go:63]: [CLOUDP-77797](https://jira.mongodb.org/browse/CLOUDP-77797) has been created.

**Review by colm-quinn — APPROVED** [review]
LGTM, thanks for the updates! Groot approves

**Pull request merged**